src/utils/warehouse/notion/milestone_formatter.rb

```diff
@@ -9,15 +11,72 @@ module Formatter
         ##
         # This class formats Notion milestone records into a standardized hash format.
         class MilestoneFormatter < Base
+          def initialize(record, external_project_id:)
+            super(record)
+            @external_project_id = external_project_id
+          end
+
           def format
             {
               external_milestone_id: extract_id,
-              name: extract_title('Name'),
-              status: extract_multi_select('Status'),
-              completion_date: extract_date('Completion Date'),
-              external_project_id: extract_relation('Project').first
+              name: extract_title('Description'),
+              status: extract_checkbox('Completed') ? 'completed' : 'incomplete',
+              completion_date: extract_date('Deadline'),
+              external_project_id: @external_project_id
             }
           end
+
+          def self.fetch_for_projects(raw_project_records, secret:, filter_body: {})
+            project_formatter = Utils::Warehouse::Notion::Formatter::ProjectFormatter
+            formatted_projects = raw_project_records.map { |record| project_formatter.new(record).format }
```

Comment:
Instead of calling the project formatter, I think we can extract the ID directly from the raw_project_records
```ruby
raw_project_records.map do |project|
  project['id']
end
```
I think it would be cleaner